packages/snap/src/utils/activitiy.ts

```diff
@@ -56,8 +56,30 @@ export async function buildNeedToNotifyContents() {
     (item) => item.watchedProfiles ?? [],
   );
 
+  const filteredProfiles: TSocialGraphResult[] = [];
+  const templeAddresses = new Set();
+  for (const profile of watchedProfiles) {
+    const filteredFollowing = [];
+    if (profile.following) {
+      for (const item of profile.following) {
```

Comment:
```
for (const profile of watchedProfiles) {
  const following = profile.following?.filter(item => {
    const { address } = item;
    if (!tempAddresses.has(address)) {
      tempAddresses.add(address);
      return true;
    }
    return false;
  });
// ...
}
```

get rid of the nested for...of loop

Reply:
I have replaced the nested for..of loops with the filter function.